cpp/open3d/core/Tensor.cpp

```diff
@@ -1081,6 +1081,19 @@ std::vector<Tensor> Tensor::NonZeroNumpy() const {
 
 Tensor Tensor::NonZero() const { return kernel::NonZero(*this); }
 
```

Comment:
The single element non-zero test is mainly added for python's `Tensor.__bool__()` operation. The `Tensor.__bool__()` is added to fix a unit test bug. Previously we have:
```python
a = Tensor::ones((5,));
assert a.sum() == 5
```
`a.sum() == 5` evaulates to `Tensor([True])`, but we can't directly assert `Tensor([True])` in this case. Therefore we need a way to evaluate a single-element Tensor as boolean. Numpy and PyTorch has the same semantics:
```python
assert np.array([True])        # Good
assert np.array([False])       # AssertionError
assert np.array([False, True]) # ValueError: cannot convert 2-element tensor to bool
```
I added some additional docs to explain this.

`Tensor::NonZero()` behaves differently since it returns `int64_t` indices. So they are differnt operations.